Pull request with the main repository

SkyL0ver opened the pull request.

Commit:
Extended interface

Commit:
Added new site

Commit:
Added HtmlUnit driver

Commit:
Moved to formats package

Commit:
Added cookie edit view

Commit:
Removed EPUB class import

Commit:
Added new output format

Commit:
Added Source listing functions

Commit:
Added new output format

Commit:
Version number

Commit:
Added HtmlUnit headless browser

Commit:
New login system

Commit:
New login system

Commit:
Build supported source string from files

Commit:
Added mature check. Added login cookies to chapter request

Commit:
Fixed missing page counter

Commit:
Added log file for errors

Commit:
Removed stacktrace print from unsupported source error

Commit:
Fixed busy icon not going away on errors

Commit:
Added cover null check

Commit:
Added asianovel

Commit:
Fixed cookies

Commit:
Removed old comment

Commit:
Add empty cookie name check

Commit:
Fixed headless grabbing not working for chapters

Commit:
Added save cookies button to login panel

Commit:
Switched thrown exceptions for setting source and stopped grabbing

Commit:
Increased version number

Commit:
Added InterruptedException throw to CLI download

Commit:
Centered new window on screen

Commit:
Removed ' as a valid encasing character for cli strings

Commit:
Updated metadata selectors

Commit:
Fixed source not being set for CLI novel object creation

Commit:
Added library checking gui indicator

Commit:
Changed save cookies button to be always visible

Commit:
Version number

Commit:
Added message to indicate if login is used

Commit:
Removed browser parameter in driver creation

Commit:
Removed obsolete driver creation

Commit:
Fixed headless driver not closing correctly

Commit:
Refactored

Commit:
Added new source

Commit:
Imports cleanup

Commit:
Comment

Commit:
Renamed Settings to Config

Commit:
Moved saveCover method from LibNovel, added covername variable

Commit:
Renamed Settings to Config

Commit:
Import cleanup

Commit:
Added static wait time of 1 second

Commit:
Added new source

Commit:
Library refactoring

Commit:
Imports cleanup

Commit:
Added mouse event to library novel title to show settings

Commit:
Added new source

Flameish committed:
Update README.md

Commit:
Added PDF output format

Commit:
Added new sources

Commit:
Version number

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Added source

Commit:
Updated selector

Commit:
Updated

Commit:
Version number

Commit:
Added default values if setting does not exist in file

Commit:
Added message null check

Commit:
Added new source

Commit:
Add new telegram settings

Commit:
Library novels try to modify previous EPUB files

Commit:
Add retry of failed chapters

Commit:
Added missing encoding

Commit:
Fixed wrong cover path

Commit:
Add new source

Commit:
Move previously downloaded file to library folder

aidanBurton commented on src/main/java/bots/Telegram.java:
```diff
@@ -0,0 +1,336 @@
+    }
+
+    private String getLimitString(int userId) {
+        String msgMaxChNovel = String.format("Max. chapters per novel: %s \n",
```
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.

Commit:
Added cookie support to manual grabbing

Commit:
Version number

Flameish committed:
Update README.md

Commit:
Removed printstack log for missing source files

Commit:
Switched font to DejaVu Sans

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Reset internal chapter numbering for each novel

Commit:
Adjust selectors

Commit:
Version number

Commit:
Add option to separate chapters into own files

Commit:
Renamed Novel field "epubFilename" to "filename"

Commit:
Version number

Commit:
Hide standard save location field if checkbox is not selected

Commit:
Settings rearranging

Flameish committed:
Update README.md

Commit:
Added

Commit:
Various small bot fixes and improvements

Commit:
Added Telegram Bot config panel

Commit:
Merge remote-tracking branch 'origin/master'

Flameish committed:
Update issue templates

Commit:
Add new source NovelBuddy

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Added blacklisted tags for duplicate chapter text

Commit:
Added new source

Commit:
Added new source

Commit:
Added new source

Commit:
Fixed images being removed because of empty block check

Commit:
Cleaned up WattPad source

Commit:
Added search, only updatable and no covers functions

Commit:
Version number

Commit:
Library GUI changes

Commit:
Added getSource method

Commit:
Added host selection filter to library

Commit:
Version number

Commit:
Adjusted empty block selector

Commit:
Added new source

Commit:
Fixed missing data

Commit:
Add new options to library novels

Commit:
Fixed desktop new chapter notifications

Commit:
Changed logging messages

Commit:
Added optional notification for finished download

Commit:
Fixed desktop new chapter notifications

Commit:
Version number

Commit:
Added variable concurrent telegram download support

Commit:
Fixed selectors

Commit:
Fixed casting error

Commit:
Changed novel window to telegram

Commit:
Added new source Syosetu

Commit:
Fixed casting error

Commit:
Switched NovelUpdates to full headless

Commit:
Switched NovelUpdates to full headless

Commit:
Small gui changes

Commit:
Updated

Commit:
Updated Library screenshot

Commit:
Added link to sources folder

Commit:
Updated

Commit:
Adjusted chapter link selector

Commit:
Added theme selection

lyndseyBeil commented on src/main/java/grabber/formats/EPUB.java:
```diff
@@ -30,27 +34,41 @@
+        if (book == null) {
+            book = new Book();
+            try {
+                book.getResources().add(new Resource(getClass().getResourceAsStream("/default.css"), "default.css"));
```
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [UI: Usage of GetResource may be unsafe if class is extended (UI_INHERITANCE_UNSAFE_GETRESOURCE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#ui-usage-of-getresource-may-be-unsafe-if-class-is-extended-ui-inheritance-unsafe-getresource).
Calling this.getClass().getResource(...) could give results other than expected if this class is extended by a class in another package.

Commit:
Added Telegram Bot option to disable image download

Commit:
Added Telegram Bot option to disable image download

Commit:
Fixed empty chapter list

Flameish committed:
Update README.md

Flameish committed:
Update README.md

Flameish committed:
Update README.md

Commit:
Adjusted metadata selectors

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Added windows folder name cleaning

Commit:
Fixed concurrent downloads updating the same progress message

Commit:
Version number

Commit:
Give name to downloaded images if filename empty

Commit:
Add headless environment check on gui startup

Commit:
Version number

Commit:
Add font loading

Commit:
Add font setting

Commit:
Add cjk font

Commit:
Add font loading

Commit:
Add font selection

Commit:
Escape HTML entities in ePub ToC and Description

Commit:
Version number

rvail committed:
maven support

Created maven pom file to make building easier.
Created gitignore file to ignore maven and ide files/dirs

rvail committed:
updated maven to build "fat" jar

Flameish committed:
Merge pull request #186 from rvail/master

Maven support

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Write log file to current path

Commit:
Add empty email host check before email client startup try

Commit:
Set caret position of chapter preview scrollbar to top

Commit:
Resolve dependency conflict

Commit:
Removed stacktrace from error message if container was not found

Commit:
Add wuxiaworldsite.co

Commit:
Add message for not found selectors when fetching chapter list

Commit:
Updated to use headless browser

Commit:
Include Intellij UI Designer in maven build

Commit:
Added Kakuyomu

Commit:
Fixed chapter container selector

Commit:
Merge remote-tracking branch 'origin/master'

posiaden123 committed:
added new site

posiaden123 committed:
added AO3, works a little different

posiaden123 committed:
deleted AO3, caught HttpExceptions

posiaden123 committed:
added 78zaotl.com

Commit:
Updated

posiaden123 committed:
fixed what I can, and what I believe is correct for the p tags. Not sure what to query for wnmtl

posiaden123 committed:
fixed post to ajax for wnmtl.org

Flameish committed:
Merge pull request #205

Adding new sites

Commit:
Rewrite Telegram bot

Commit:
Fixed description selector

Commit:
Fixed formatting

Commit:
Version number

Commit:
Add site

Commit:
Improved error feedback if old book file couldn't be accessed

Commit:
Moved exception block

Commit:
Version number

Commit:
Add chapter title customization

Flameish committed:
Update issue templates

Flameish committed:
Update issue templates

Commit:
Removed the 'Remove styling' option

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Added Betwixted Translations

Commit:
Added NovelHD

Commit:
Added WuxiaCity

Commit:
Added Very Active Fan Translations

Commit:
Removed old Active Translations source

Commit:
Rename to Active Translations

Commit:
Fixed chapter body selector

Commit:
Fixed Novel tab icon in settings

Commit:
Increased version number

Flameish committed:
Update README.md

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Updated version number

Commit:
Updated

Commit:
Added HTTP error code

Commit:
Library remembers changed novel name now

Commit:
Increased window size

Commit:
Added custom filename template

Commit:
Fixed missing extension when try to read old file

Commit:
Fixed wrong icon

Commit:
Version number

Commit:
Fix issue where url request for an image will instead response with a html document.

Flameish committed:
Merge pull request #260 from mastertea/master

Fix issue where url request for an image will instead response with a html document.

Commit:
Adjusted table of contents page

Commit:
Fixed WNMTL

Commit:
Hide locked chapters

Commit:
Merge remote-tracking branch 'origin/master'

dependabot[bot] committed:
Bump jsoup from 1.13.1 to 1.14.2

Bumps [jsoup](https://github.com/jhy/jsoup) from 1.13.1 to 1.14.2.
- [Release notes](https://github.com/jhy/jsoup/releases)
- [Changelog](https://github.com/jhy/jsoup/blob/master/CHANGES)
- [Commits](https://github.com/jhy/jsoup/compare/jsoup-1.13.1...jsoup-1.14.2)

---
updated-dependencies:
- dependency-name: org.jsoup:jsoup
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Flameish committed:
Merge pull request #273 from Flameish/dependabot/maven/org.jsoup-jsoup-1.14.2

Bump jsoup from 1.13.1 to 1.14.2

Flameish committed:
Update README.md

Flameish committed:
Update README.md

Commit:
Fixed #297

Commit:
Added the option to block specific telegram users.

Commit:
Version number increase

Commit:
Fixed -getImages CLI command

Commit:
Add TelegramUser field to User

Commit:
Add html meta tags

Commit:
Add UUID to Telegram EPUB files

Commit:
Updated Boxnovel.org to Novelpoki.com

Commit:
Merge remote-tracking branch 'origin/master'
